logs/logs.go

```diff
@@ -106,7 +108,15 @@ func (l *LogAgent) Run(ctx context.Context) {
 						log.Printf("E! [logagent] Failed to find destination %v for log source %v/%v(%v) ", dname, src.Group(), src.Stream(), src.Description())
 						continue
 					}
-					dest := backend.CreateDest(src.Group(), src.Stream(), src.Retention())
+					retention := src.Retention()
+					if retention > 0 && l.retentionAlreadySet[src.Group()] == true {
```

Comment:
```suggestion
					if retention > 0 && l.retentionAlreadySet[src.Group()] {
```
Don't need to assert true == true